### PR TITLE
Move agent log folder on Mac from /var/log/datadog to /opt/datadog-ag…

### PR DESCRIPTION
**Pull request by ghost**
…ent/logs. A link will be created at /var/log/datadog pointing to /opt/datadog-agent/logs to maintain the compatibility.

### What does this PR do?

As title, move log location on Mac.

### Motivation

Some Mac OS releases purge /var/log folder. Agent is run under less privileged account and cannot recreate log folder under /var, so it will not start after certain OS upgrade.

This commit changes preinst script to create a different log folder and create a link at the old location.

Agent (Datadog / process / trace agent) code is also changed to use the new location.

### Additional Notes

Uninstallation on Mac is mostly manual per https://docs.datadoghq.com/agent/faq/how-do-i-uninstall-the-agent/?tab=agentv6v7 

I will change doc to reflect the change.

### Describe your test plan
On Mac
Test 1: Install old build then install this new build
Test 2: Remove /var/log/datadog folder then run datadog agent
Test 3: Remove /var/log folder (simulate OS upgrade)  then install the new build or run datadog agent.
Make sure agent works in all cases.

### CODE REVIEW

**Review by gbbr — APPROVED** [review]

File: omnibus/package-scripts/agent/postinst [hunk 1]
```diff
@@ -131,9 +131,13 @@ else
     CONF_DIR=$INSTALL_DIR/etc
     RUN_DIR=$INSTALL_DIR/run
 
+    # On Mac, the real log folder is located under the user profile folder, because some MacOS versions deletes
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/postinst:134]: `under the user profile` is not really the case.

File: omnibus/package-scripts/agent/preinst [hunk 2]
```diff
@@ -120,14 +120,52 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
...
+    mkdir -p $REAL_LOG_DIR
 
     CONF_DIR="$INSTALL_DIR/etc"
     APP_DIR="/Applications/Datadog Agent.app"
 
-    LOG_FILE="$LOG_DIR/preinstall.log"
+    LOG_FILE="$REAL_LOG_DIR/preinstall.log"
     exec > $LOG_FILE 2>&1
 
+    if [ -d $LOG_DIR ]; then
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/preinst:137]: Is this code path used on both new installs and upgrades? It should, but I'm not sure now.

**Review by albertvaka — COMMENTED** [review]
LGTM but I left one question

File: omnibus/package-scripts/agent/preinst (hunk 2 above)
Comment (mikezhu-dd) [reply omnibus/package-scripts/agent/preinst:137]: It's used in both cases. But Mac doesn't track pkg file, unlike the rpm on Linux, so there's no check for upgrade.

File: omnibus/package-scripts/agent/postinst (hunk 1 above)
Comment (mikezhu-dd) [reply omnibus/package-scripts/agent/postinst:134]: I'll change the comment. Thanks for catching that.

**Review by albertvaka — APPROVED** [review]

**Pull request merged**